### PR TITLE
test: improve HTLC tests, add docs, raise exception on invalid unlock_secret format

### PR DESCRIPTION
**Pull request by c0nsol3**
Added docstrings, modified tests with more explanation and realistic examples, raising an exception in case `unlock_secret` is in wrong format.

Resolves https://github.com/ArkEcosystem/python-crypto/issues/122

### CODE REVIEW

File: crypto/transactions/serializers/htlc_claim.py [hunk 1]
```diff
@@ -9,7 +10,10 @@ class HtlcClaimSerializer(BaseSerializer):
 
     def serialize(self):
         self.bytes_data += unhexlify(self.transaction['asset']['claim']['lockTransactionId'])
-        unlock_secret = hexlify(self.transaction['asset']['claim']['unlockSecret'].encode())
```
Comment (c0nsol3) [inline crypto/transactions/serializers/htlc_claim.py:12]: this was an unnecessary hexlify/unhexlify that ends up results in the same thing

File: tests/transactions/builder/test_htlc_claim.py [hunk 2]
```diff
@@ -1,58 +1,91 @@
...
     lock_transaction_id = '943c220691e711c39c79d437ce185748a0018940e1a4144293af9d05627d2eb4'
 
-    # This should be the hashed secret used for HTLC Lock transaction
-    unlock_secret = 'my secret that should be 32bytes'
+    # Secret required to unlock and claim funds from an HTLC. This secret should is revealed to you
+    # on the blockchain which you can then use to claim funds.
+    unlock_secret = '6434323233626639336532303235303561366135303134323161383864396661'
 
     transaction = HtlcClaim(lock_transaction_id, unlock_secret)
-    transaction.set_type_group(TRANSACTION_TYPE_GROUP.CORE)
```
Comment (c0nsol3) [inline tests/transactions/builder/test_htlc_claim.py:18]: these ones are not needed as the type group is set directly in `HtlcClaim`'s init class.

File: tests/transactions/builder/test_htlc_lock.py [hunk 3]
```diff
@@ -103,8 +109,20 @@ def test_htlc_lock_transaction_custom_fee():
...
     assert transaction_dict['asset']['lock']['expiration']['value'] == 1573455822
 
     transaction.schnorr_verify()  # if no exception is raised, it means the transaction is valid
+
+
+def _generate_secret_hash():
+  secret = "super secret code that must be unique and entirely random"
+
+  # secret code is a key ingredient used to unlock the htlc lock. To generate an unlock code one
+  # must "hexify" it: `unlock_secret = hexlify(secret_h.encode()).decode()``
```
Comment (c0nsol3) [inline tests/transactions/builder/test_htlc_lock.py:123]: having this explanation somewhere in a test would have saved me a lot of time. Trying to figure out why my transaction was being rejected and how to properly construct an `unlock_secret` was annoying.

**Comment by c0nsol3** [comment]
I'll have to fix two tests that aren't passing after this change:
- `tests/transactions/serializers/test_htlc_claim.py`
- `tests/transactions/deserializers/test_htlc_claim.py`

**Review by faustbrian — APPROVED** [review]

**Pull request merged**